### DISCUSSION TITLE
Height placement helpers when using shift/control keys

### DISCUSSION THREAD

**Discussion by sta5is** [OpenRCT2/OpenRCT2#22095] [Feature requests]
OS: Windows 8.1
Version: 0.1.3
Commit/Build: 617372f

Something similar to this would be very helpful when building things vertically (taken from Themeparkitect):
![tumblr_inline_o53pwfnmwp1si3zx7_400](https://user-images.githubusercontent.com/26368692/40591689-7f73e34a-61ca-11e8-9e52-9ad7a180d341.gif)

Building in an isometric environment can make things difficult when placing objects at a higher level. I think something like this would greatly ease this issue. While holding shift/control key, if the object is raised above the surface tile, the following happens:
- A perforated line drawn from the surface tile to the elevated object. Each segment of the line indicates one base-height unit.
- A numeric value showing the object's height. This can be the in-game height unit or the base-height value.

**Comment by jensj12** [comment]
There is already something in the game that helps when holding ctrl/shift. Have you tried the virtual floor? If you don't see it, enable it in the options.
Anyways, the virtual floor can be quite disrupting. This can be a good replacement for those who deliberately disabled the virtual floor.

**Comment by Gymnasiast** [comment]
Closing as the virtual floor is pretty much what OP described.

**Comment by sta5is** [comment]
While the virtual floor is useful, its utility becomes disruptive when an area is populated with lots of objects. The bounding lines become less and less informative especially when building detailed areas with disabled clearance. I hope this idea can be reconsidered.

**Comment by AaronVanGeffen** [comment]
We might be able to add a height label to an object as it's being placed. Not sure how easily that can be added to the current rendering routines, though.

This would also be solved if we implemented #3419 as a togglable option.

**Comment by ocalhoun6** [comment]
A number could be placed on the corner of the virtual floor, indicating the height that the virtual floor is currently at. That could be helpful.

**Comment by sta5is** [comment]
Sorry to bump this old request. I'm currently building a very vertical park and was wondering if it's possible to revisit this idea? I understand that you guys have more important tasks going on but this will be very helpful to me and I'm sure for others as well. thank you

**Comment by duncanspumpkin** [comment]
Feel free to try implement it.
